sklearn/feature_extraction/dict_vectorizer.py

```diff
@@ -94,11 +95,34 @@ class DictVectorizer(BaseEstimator, TransformerMixin):
     """
 
     def __init__(self, dtype=np.float64, separator="=", sparse=True,
-                 sort=True):
+                 sort=True, n_features=None):
+        if n_features is not None:
+            self._validate_params(n_features)
```

Comment:
We conventionally do all validation in `fit`. There are reasons for this. See [Glossary: parameter](http://scikit-learn.org/dev/glossary.html#term-parameter)